no-empty-character-class rule not working well with unicode flag (ES6)

The `no-empty-character-class` rule gives a very confusing error whenever it encounters a regular expression using the ES6 `u` flag. (for unicode support)

To be clear, my regular expression does _not_ have an empty character class, and if I remove the `u` flag, it is not treated as an error anymore.

It looks like this is caused by the fact that the only recognized flags are `gimy`. (see [code](https://github.com/eslint/eslint/blob/master/lib/rules/no-empty-character-class.js#L21)) I presume adding `u` to the list would fix this problem, but not sure if that should be conditionally added with ES6 support.

This is using the latest version of ESLint. (ie: `0.22.1`)

Here's a distilled version:

**.eslintrc**

```js
{
  "ecmaFeatures": {
    "regexUFlag": true
  },
  "rules": {
    "no-empty-character-class": 2
  }
}
```

**test.js**

```js
var r1 = /^[^\s]+$/u;
var r2 = /^[^\s]+$/;
```

**CLI**

```
$ eslint --reset test.js

test.js
  1:9  error  Empty class  no-empty-character-class

✖ 1 problem (1 error, 0 warnings)

```

I'm expecting that both regular expressions would pass the linter, but it flags the first one as an error. (the only difference is the `u` flag)

--- a/lib/rules/no-empty-character-class.js
+++ b/lib/rules/no-empty-character-class.js
@@ -21,7 +21,7 @@ plain-English description of the following regexp:
 4. `[gimy]*`: optional regexp flags
 5. `$`: fix the match at the end of the string
 */
-var regex = /^\/([^\\[]|\\.|\[([^\\\]]|\\.)+\])*\/[gimy]*$/;
+var regex = /^\/([^\\[]|\\.|\[([^\\\]]|\\.)+\])*\/[gimuy]*$/;
 
 //------------------------------------------------------------------------------
 // Rule Definition

--- a/tests/lib/rules/no-empty-character-class.js
+++ b/tests/lib/rules/no-empty-character-class.js
@@ -28,7 +28,8 @@ eslintTester.addRuleTest("lib/rules/no-empty-character-class", {
         "var foo = /[[]/;",
         "var foo = /[\\[a-z[]]/;",
         "var foo = /[\\-\\[\\]\\/\\{\\}\\(\\)\\*\\+\\?\\.\\\\^\\$\\|]/g;",
-        "var foo = /\\s*:\\s*/gim;"
+        "var foo = /\\s*:\\s*/gim;",
+        { code: "var foo = /[\\]]/uy;", ecmaFeatures: { regexUFlag: true, regexYFlag: true } }
     ],
     invalid: [
         { code: "var foo = /^abc[]/;", errors: [{ message: "Empty class.", type: "Literal"}] },

